presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/mysql/FunctionNamespaceDao.java

```diff
@@ -51,13 +51,14 @@
 
     @SqlUpdate("CREATE TABLE IF NOT EXISTS <sql_functions_table> (\n" +
             "  id bigint(20) NOT NULL AUTO_INCREMENT,\n" +
-            "  function_id varchar(700) NOT NULL,\n" +
+            "  function_id_hash varchar(128) NOT NULL,\n" +
+            "  function_id text NOT NULL,\n" +
```

Comment:
At this point i think if you store parameters as separate list of name and list of type you can remove `function_id`. This is just a waste of storage space.

Reply:
Sure, I can do that separately. Let's work on merging this fix first. Thanks!

Reply:
If you plan to do that, do it in this PR. We want to minimize schema changes across releases. This creates a lot of pain to early adopters.

Reply:
So I made an attempt to this and come to think that this is not a worthwhile effort.
- A function_id is a (catalog_name, schema_name, function_name, parameter_types) tuple.
- We already have separate columns for catalog_name, schema_name, and function_name, but we don't store parameter_types.
- It's not removing `function_id`, it is essentially replacing `function_id` with a new column `parameter_types` (because it is never needed by itself, only needed for filtering, we can always use the `parameters` column).
- If we store `function_id`, duplication is only catalog/schema/function, which are small, fixed length, and given we also have small amount of entries, this is tiny.
- Downside of this replacement 1 filter clause / 1 parameter to become 4 clause / 4 parameters everyone, which makes the code less readability. I don't think it's worthwhile to trade a few MB of table sizes with readability/maintainaility.